test/e2e/validation_test.go

```diff
@@ -742,13 +790,69 @@ func (tc *testContext) getOperatorConditionName() (string, error) {
 	return "", fmt.Errorf("unable to get operatorCondition name from namespace %s", wmcoNamespace)
 }
 
+// testNodeAnnotations tests that all required annotations are on each Windows node
+func (tc *testContext) testNodeAnnotations(t *testing.T) {
+	for _, node := range gc.allNodes() {
+		t.Run(node.GetName(), func(t *testing.T) {
+			annotations := []string{nc.HybridOverlaySubnet, nc.HybridOverlayMac, metadata.VersionAnnotation,
+				nc.PubKeyHashAnnotation}
+			for _, annotation := range annotations {
+				assert.Contains(t, node.Annotations, annotation, "node missing expected annotation: %s", annotation)
+			}
+
+			usernameCorrect, err := tc.usernameAnnotationCorrect(&node)
+			require.NoError(t, err)
+			assert.True(t, usernameCorrect)
+
+			pubKey, err := tc.pubKeyAnnotationCorrect(&node)
+			require.NoError(t, err)
+			assert.True(t, pubKey)
+		})
+	}
+}
+
+// usernameAnnotationCorrect checks that the username annotation value is decipherable and correct
+func (tc *testContext) usernameAnnotationCorrect(node *core.Node) (bool, error) {
```

Comment:
If you want to go with this sort of naming I suggest `isUsernameAnnotationCorrect()` though I prefer `checkUserNameAnnotation()`